geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/IdExtractingVisitor.scala

```diff
@@ -42,13 +42,10 @@ class IdExtractingVisitor extends FilterVisitor {
     (andOption(yes).orNull, andOption(no).orNull)
   }
 
-  override def visit(f: Not, data: AnyRef): AnyRef = {
-    val (yes, no) = f.getFilter.accept(this, data).asInstanceOf[(Filter, Filter)]
-    (if (yes == null) null else ff.not(yes), if (no == null) null else ff.not(no))
```

Comment:
SimplifyingFilterVisitor?  https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/filter/visitor/SimplifyingFilterVisitor.java

I thought we were using it...